v2/websocket_service.go

```diff
@@ -9,33 +9,43 @@ import (
 )
 
 // Endpoints
-const (
-	baseWsMainURL          = "wss://stream.binance.com:9443/ws"
-	baseWsTestnetURL       = "wss://testnet.binance.vision/ws"
-	baseCombinedMainURL    = "wss://stream.binance.com:9443/stream?streams="
-	baseCombinedTestnetURL = "wss://testnet.binance.vision/stream?streams="
-)
-
 var (
 	// WebsocketTimeout is an interval for sending ping/pong messages if WebsocketKeepalive is enabled
 	WebsocketTimeout = time.Second * 60
 	// WebsocketKeepalive enables sending ping/pong messages to check the connection stability
 	WebsocketKeepalive = false
+	// default ws baseURL as protocol+host[+port]
+	baseURL = "wss://stream.binance.com:9443"
+	// default testnet ws baseURL as protocol+host[+port]
+	baseTestnetURL = "wss://testnet.binance.vision"
+	// derived from baseURL
+	baseWsMainURL       string
+	baseCombinedMainURL string
 )
 
-// getWsEndpoint return the base endpoint of the WS according the UseTestnet flag
-func getWsEndpoint() string {
+func init() {
 	if UseTestnet {
```

Comment:
Do not use init with a mutable global. It's likely to run before `UseTestnet` is set.

Reply:
hi @asamere , check this comment, I agree with @norunners . `UseTestnet` could be updated after `init`.